src/Task/Builder/ChainedBuilder.php

```diff
@@ -38,7 +43,7 @@ public function __construct(
      */
     public function build(PropertyAccessorInterface $propertyAccessor, array $options = []): TaskInterface
     {
-        $chainedTask = new ChainedTask($options['name'], ...array_map(function (array $task) use ($propertyAccessor): TaskInterface {
+        $chainedTask = new ChainedTask($options['name'], ... $this->orchestrator->sort($options['execution_mode'], array_map(function (array $task) use ($propertyAccessor): TaskInterface {
```

Comment:
I'm not sure about the structure here, for me, `execution_mode` is a new attribute (with getter/setter in `ChainedTask` so IMHO, we should update the signature according to: 

```php
public function __construct(string $name, string $executionMode, TaskInterface ...$tasks)
```

Reply:
So it will be the ChainedTask that will be in charge of ordering the tasks ?

Reply:
Regarding yesterday discussion, could we add more tests for this builder to ensure that tasks are correctly sorted once created?

Maybe with multiple policies? I'm starting to think that some (like FIFO) can thrown error if we don't have the required attributes).

Reply:
will do this

Reply:
i have pushed some tests, actually only the `testBuilderCanSortTasksUsingDeadline`failed

Reply:
I think it is important to improve the validation of the configuration, the problem must be identical on simple tasks if the option is not defined
https://github.com/Guikingone/SchedulerBundle/blob/main/src/DependencyInjection/SchedulerBundleConfiguration.php#L67-L92

Reply:
Agree, the main issue is that we have several options and listing each can be hard 🤔

Reply:
https://github.com/Guikingone/SchedulerBundle/blob/0efd88bdd39c04dda6f9f821465f70f1305ea15a/src/SchedulePolicy/DeadlinePolicy.php#L27-L29

Shouldn't we throw an exception if the getExecutionRelativeDeadline() function is null instead of continue? 
Because if it's null in any case it will crash in the uasort function.

Reply:
ok sorry for all those questions :(

I saw that the arrivalTime and executionStartTime are set only on the master task
https://github.com/Guikingone/SchedulerBundle/blob/0efd88bdd39c04dda6f9f821465f70f1305ea15a/src/Worker/Worker.php#L233-L234

would it be interesting to set this on subtasks ?

Reply:
`continue` allows us to continue the process without sorting the current task, throwing an exception will force the whole process to crash, both approaches are "valid" IMHO, what's your arguments for throwing an exception?

Reply:
`continue` continues the process but does not exclude the task from the array used in the `uasort` callback function which will also crash. why continue a process when the sorting cannot succeed.

but it is true that adding an exception is not necessarily useful because it will crash in both cases.

no more agruments :(

Reply:
Yes, you're right, I'm gonna fix this behaviour 🙂

Reply:
`DeadlinePolicy` improved.